Frameworks/CoreMotion/CMMotionManager.mm

```diff
@@ -37,123 +42,108 @@
 @property (readwrite, nonatomic, getter=isAccelerometerActive) BOOL accelerometerActive;
 @property (readwrite, nonatomic, getter=isAccelerometerAvailable) BOOL accelerometerAvailable;
 @property (readwrite) CMAccelerometerData* accelerometerData;
 
-@property WDSGyrometer* gyrometer;
-@property EventRegistrationToken gyrometerToken;
+@property winrt::event_token gyrometerToken;
 @property NSOperationQueue* gyroQueue;
 @property (readwrite, nonatomic, getter=isGyroActive) BOOL gyroActive;
 @property (readwrite, nonatomic, getter=isGyroAvailable) BOOL gyroAvailable;
 @property (readwrite) CMGyroData* gyroData;
 
-@property WDSMagnetometer* magnetometer;
-@property EventRegistrationToken magnetometerToken;
+@property winrt::event_token magnetometerToken;
 @property NSOperationQueue* magnetometerQueue;
 @property (readwrite, nonatomic, getter=isMagnetometerActive) BOOL magnetometerActive;
 @property (readwrite, nonatomic, getter=isMagnetometerAvailable) BOOL magnetometerAvailable;
 @property (readwrite) CMMagnetometerData* magnetometerData;
 
 // The orientation sensors returns rotation matrix and quaternion.
 // We will use the rotation matrix to calculate pitch, roll and yaw values.
-@property WDSOrientationSensor* orientation;
-@property EventRegistrationToken deviceMotionToken;
+@property winrt::event_token deviceMotionToken;
 @property NSOperationQueue* deviceMotionQueue;
 @property (readwrite, nonatomic, getter=isDeviceMotionActive) BOOL deviceMotionActive;
 @property (readwrite, nonatomic, getter=isDeviceMotionAvailable) BOOL deviceMotionAvailable;
 @property (readwrite) CMDeviceMotion* deviceMotion;
 
 @end
 
-@implementation CMMotionManager
+@implementation CMMotionManager {
+    TrivialDefaultConstructor<Accelerometer> _accelerometer;
+    TrivialDefaultConstructor<Gyrometer> _gyrometer;
+    TrivialDefaultConstructor<Magnetometer> _magnetometer;
+    TrivialDefaultConstructor<OrientationSensor> _orientation;
+}
 
 - (instancetype)init {
     if (self = [super init]) {
         // GetTickCount64() returns the number of milliseconds passed since device boot
-        _bootTime = static_cast<double>(GetTickCount64());
-
-        FILETIME fileTime;
-        GetSystemTimeAsFileTime(&fileTime);
-
-        ULARGE_INTEGER nowTime;
-        nowTime.LowPart = fileTime.dwLowDateTime;
-        nowTime.HighPart = fileTime.dwHighDateTime;
-
-        // On iOS the timestamps represent the number of seconds passed since the device booted while
-        // WinRT returns the number of 100-nanosecond intervals passed after midnight on January 1, 1601.
-        // Hence we calculate the bootTime as number of 100-nanosecond intervals after midnight on January 1, 1601
-        // till device boot and then we can simply subtract this value from each readings' timestamp.
-        _bootTime = static_cast<double>(nowTime.QuadPart) - (_bootTime * c_millisecondToHundredNanoseconds);
+        ULONGLONG ticks = GetTickCount64();
```

Comment:
Normally yes, but GetTickCount64 is declared as returning ULONGLONG, so I think it's best to conform.

---
In reply to: [111271794](https://github.com/Microsoft/WinObjC/pull/2505#discussion_r111271794) [](ancestors = 111271794)